components/auth/org.wso2.carbon.auth.scim.rest.api/src/main/java/org/wso2/carbon/auth/scim/rest/api/impl/GroupsApiServiceImpl.java

```diff
@@ -9,43 +13,91 @@
+        UserManager userManager;
+        try {
+            userManager = SCIMManager.getInstance().getCarbonAuthUserManager();
+            GroupResourceManager groupResourceManager = new GroupResourceManager();
+            SCIMResponse scimResponse = groupResourceManager.delete(id, userManager);
+            return ApiServiceUtils.buildResponse(scimResponse);
+        } catch (AuthUserManagementException e) {
+            LOG.error("Error in initializing the CarbonAuthUserManager");
+        }
+        return Response.serverError().build();
     }
     @Override
     public Response groupsIdGet(String id
   ,Request request) throws NotFoundException {
-        // do some magic!
-        return Response.ok().entity(new ApiResponseMessage(ApiResponseMessage.OK, "magic!")).build();
+        UserManager userManager;
+        try {
+            userManager = SCIMManager.getInstance().getCarbonAuthUserManager();
+            GroupResourceManager groupResourceManager = new GroupResourceManager();
+            SCIMResponse scimResponse = groupResourceManager.get(id, userManager,  null, null);
+            return ApiServiceUtils.buildResponse(scimResponse);
+        } catch (AuthUserManagementException e) {
+            LOG.error("Error in initializing the CarbonAuthUserManager");
+        }
+        return Response.serverError().build();
     }
     @Override
     public Response groupsIdPut(String id, GroupDTO body
   ,Request request) throws NotFoundException {
-        // do some magic!
-        return Response.ok().entity(new ApiResponseMessage(ApiResponseMessage.OK, "magic!")).build();
+        UserManager userManager;
+        try {
+            userManager = SCIMManager.getInstance().getCarbonAuthUserManager();
+            GroupResourceManager groupResourceManager = new GroupResourceManager();
+            SCIMResponse scimResponse = groupResourceManager.updateWithPUT(id, body.toString(), userManager,  null, null);
+            return ApiServiceUtils.buildResponse(scimResponse);
+        } catch (AuthUserManagementException e) {
+            LOG.error("Error in initializing the CarbonAuthUserManager");
+        }
+        return Response.serverError().build();
     }
     @Override
     public Response groupsPost(GroupDTO body
   ,Request request) throws NotFoundException {
-        // do some magic!
-        return Response.ok().entity(new ApiResponseMessage(ApiResponseMessage.OK, "magic!")).build();
+        UserManager userManager;
+        try {
+            userManager = SCIMManager.getInstance().getCarbonAuthUserManager();
+            GroupResourceManager groupResourceManager = new GroupResourceManager();
+            SCIMResponse scimResponse = groupResourceManager.create(body.toString(), userManager,  null, null);
+            return ApiServiceUtils.buildResponse(scimResponse);
+        } catch (AuthUserManagementException e) {
+            LOG.error("Error in initializing the CarbonAuthUserManager");
```

Comment:
Exception has not been logged. Need to log exception as follows,

log.error("Message", e);